test/index.ts

```diff
@@ -129,6 +131,28 @@ describe('monocle-ts', () => {
     const lenses = lensesFromProps(Person.props)
     assert.strictEqual(lenses.age.get({ name: 'Giulio', age: 43 }), 43)
   })
+
+  describe('ISOType', () => {
+    type A = {
+      a: string
+    }
+    const a: A = { a: 'x' }
+    type B = {
+      b: string
+    }
+    const b: B = { b: 'x' }
+
+    const ABiso = new m.Iso<A, B>(n => ({ b: n.a }), n => ({ a: n.b }))
+    const ABType = ISOType('t', ABiso, (n): n is B => n.hasOwnProperty('b') && typeof n['b'] === 'string')
```

Comment:
...so here the `is` function should check that `n` is an object in the first place

Reply:
indeed